### PR TITLE
Feature/add paypal checkout

### PR DESCRIPTION
**Pull request by emily40830**

### CODE REVIEW

**Comment by emily40830** [comment]
[(5)【金流】paypal串接[2]](https://trello.com/c/CfU1fRvV/2199-5%E3%80%90%E9%87%91%E6%B5%81%E3%80%91paypal%E4%B8%B2%E6%8E%A52)

File: src/helpers/translation.ts [hunk 1]
```diff
@@ -1622,6 +1635,14 @@ export const codeMessages = defineMessages({
     id: 'code.E_PAYMENT_GATEWAY',
     defaultMessage: 'E_PAYMENT_GATEWAY',
   },
+  E_SETUP_PAYPAL: {
+    id: 'code.E_SETUP_PAYPAL',
+    defaultMessage: 'paypal環境未設定',
+  },
```
Comment (kkshyu) [inline src/helpers/translation.ts:1641]: NITS: `E_PAYPAL_SETUP` is better

File: src/helpers/translation.ts [hunk 2]
```diff
@@ -1622,6 +1635,14 @@ export const codeMessages = defineMessages({
...
+  E_PAYPAL_: {
+    id: 'code.E_PAYPAL_',
+    defaultMessage: 'E_PAYPAL_',
+  },
```
Comment (kkshyu) [inline src/helpers/translation.ts:1645]: what's this?

File: src/pages/PaymentPage.tsx [hunk 3]
```diff
@@ -47,9 +46,16 @@ const usePayForm = (paymentNo: number) => {
             headers: { authorization: `Bearer ${authToken}` },
           },
         )
-        .then(({ data: { code, result } }) => {
+        .then(({ data: { code, message, result } }) => {
           if (code === 'SUCCESS') {
             switch (result.gateway) {
+              case 'paypal':
+                if (result.html) {
+                  window.location.href = result.html
```
Comment (kkshyu) [inline src/pages/PaymentPage.tsx:54]: why not using `setPayForm` like `spgateway` and `parenting`

**Review by kkshyu — CHANGES_REQUESTED** [review]
plz learn how to solve the conflict

**Review by kkshyu — APPROVED** [review]

File: src/helpers/translation.ts (hunk 1 above)
Comment (kkshyu) [reply src/helpers/translation.ts]: i see. we can discuss about this in the code review meeting.
i prefer E_NAMESPACE_ACTION pattern

**Pull request merged**